### PR TITLE
Move optional/required condition filtering back to check_state()

### PR DESCRIPTION
**Pull request by willbarton**
755a11290c70f7048f479525a5593d3554cc0ae7 did some refactoring of when optional and required conditions are filtered. This has resulted in some issues where conditions aren't being correctly evaluated. 

Thanks to @chosak, the issue is [`conditions` is modified on the `Flag` object after `__init__` is called](https://github.com/cfpb/django-flags/blob/3481867933e1dc84e7cbddf53954c80ac8af0898/flags/sources.py#L186), but `optional_conditions` and `required_conditions` are never updated after `__init__`. 

This seems like it was a false optimization. This PR reverts it to fix the issue.

### CODE REVIEW

File: flags/sources.py [hunk 1]
```diff
@@ -33,22 +33,23 @@ class Flag(object):
...
     def __eq__(self, other):
         """ There can be only one feature flag of a given name """
         return other.name == self.name
 
     def check_state(self, **kwargs):
         """ Determine this flag's state based on any of its conditions """
-        if (len(self.optional_conditions) == 0
-                and len(self.required_conditions) == 0):
+        optional_conditions = [
+            c for c in self.conditions if not c.required
```
Comment (chosak) [inline flags/sources.py]: Shouldn't these lines refer to `self.__conditions`? Is there still a `self.conditions`?

Comment (willbarton) [reply flags/sources.py]: Oh, oops. that was a separate change I meant to revert. `self.conditions` should be `self.conditions`!

Comment (willbarton) [reply flags/sources.py]: Fixed in 5c275d8.

**Comment by chosak** [comment]
While Travis is bogged down, it sure would be nice to have a test that tests this specific problem. 🐴

**Comment by willbarton** [comment]
Forgot to push this yesterday — test added in bb82064.

**Pull request merged**